app/celery/research_mode_tasks.py

```diff
@@ -1,45 +1,43 @@
-import json
 import random
 from datetime import datetime, timedelta
 
 from flask import current_app
 from notifications_utils.s3 import s3upload
-from requests import HTTPError, RequestException, request
 
-from app import notify_celery
+from app import create_uuid, notify_celery
+from app.aws.mocks import (
+    ses_hard_bounce_callback,
+    ses_notification_callback,
+    ses_soft_bounce_callback,
+    sns_failed_callback,
+    sns_s3_callback,
+    sns_success_callback,
+)
 from app.aws.s3 import file_exists
 from app.celery.process_ses_receipts_tasks import process_ses_results
+from app.celery.process_sns_receipts_tasks import process_sns_results
 from app.config import QueueNames
-from app.models import SMS_TYPE
 
-temp_fail = "7700900003"
-perm_fail = "7700900002"
-delivered = "7700900001"
+last_digit_temp_fail = "3"
+last_digit_perm_fail = "2"
+last_digit_delivered = "1"
 
 delivered_email = "delivered@simulator.notify"
 perm_fail_email = "perm-fail@simulator.notify"
 temp_fail_email = "temp-fail@simulator.notify"
 
 
-def send_sms_response(provider, reference, to):
-    headers = {"Content-type": "application/x-www-form-urlencoded"}
-    body = firetext_callback(reference, to)
-    # to simulate getting a temporary_failure from firetext
-    # we need to send a pending status updated then a permanent-failure
-    if body["status"] == "2":  # pending status
-        make_request(SMS_TYPE, provider, body, headers)
-        # 1 is a declined status for firetext, will result in a temp-failure
-        body = {
-            "mobile": to,
-            "status": "1",
-            "time": "2016-03-10 14:17:00",
-            "reference": reference,
-        }
-
-    make_request(SMS_TYPE, provider, body, headers)
+def send_sms_response(provider, to, reference=None):
+    if not reference:
+        reference = str(create_uuid())
```

Comment:
```suggestion
    reference = reference or str(create_uuid())
```
feel free to not do it